src/views/AddItem.jsx

```diff
@@ -25,18 +26,25 @@ export function AddItem() {
 		setFormData((formData) => ({ ...formData, [name]: value }));
 	};
 
+	const itemNames = data.map((item) =>
+		item.name.toLowerCase().replace(/[^a-z0-9]/gi, ''),
+	);
 	// if no item name provided, gives user feedback to add item
 	// if item name provided, adds item to user's list and gives user success feedback
 	const handleSubmit = (e) => {
 		e.preventDefault();
 		if (itemName === '') {
-			alert('Please enter the name of your item.');
+			setMessage('Please enter the name of your item');
+		} else if (
+			itemNames.includes(itemName.toLowerCase().replace(/[^a-z0-9]/gi, ''))
+		) {
+			setMessage(`You already have ${itemName} on your list`);
```

Comment:
This looks really great a feels more user friendly than the alert.

Reply:
This is great! If we have time, I'd love to see this implement elsewhere in the project. One todo for later: right now, the text is red regardless of the feedback. At some point we may want to add a second argument to the alert to give it a type eg) `setMessage({type: 'error', msg: 'You already...})` or `setMessage({type: 'success', msg:' ${itemName} added to list'})`

Reply:
Red inline styling is just temporary, lol; I'm sure we'll remove that and handle styling with css in our final stage.

Reply:
Good comment about the text color and agree with you both here. Red generally indicates warning or error. Prioritizing as a todo for later is a great call!